tap_hubspot/stream.py

```diff
@@ -73,10 +74,13 @@ def do_sync(self, state):
                         if prev_bookmark < new_bookmark:
                             state = self.__advance_bookmark(state, prev_bookmark)
                             prev_bookmark = new_bookmark
-                    if self.tap_stream_id == "contacts_events":
-                        prev_bookmark = end_date
+                    if self.tap_stream_id in ["contacts_events"]:
+                        prev_bookmark = event_state["contacts_end_date"]
 
-                    return self.__advance_bookmark(state, prev_bookmark)
+                    return (
+                        self.__advance_bookmark(state, prev_bookmark),
```

Comment:
As far I can see, you don't need to return this `self.__advance_bookmark(state, prev_bookmark)` - you don't use it in the calling `forloop` in `__init__`

Reply:
I think this way the state can be appended. {bookmarks:{companies: state}}